javascript/modules/client/test/integration/client-erc20.test.ts

```diff
@@ -582,31 +584,106 @@ describe("Client", () => {
+        expect(typeof proposal.dao.address).toBe("string");
+        expect(proposal.dao.address).toMatch(/^0x[A-Fa-f0-9]{40}$/i);
+        expect(typeof proposal.dao.name).toBe("string");
+        expect(typeof proposal.creatorAddress).toBe("string");
+        expect(proposal.creatorAddress).toMatch(/^0x[A-Fa-f0-9]{40}$/i);
+        expect(typeof proposal.metadata.title).toBe("string");
+        expect(typeof proposal.metadata.summary).toBe("string");
+        expect(proposal.startDate instanceof Date).toBe(true);
+        expect(proposal.endDate instanceof Date).toBe(true);
+        // result
+        expect(typeof proposal.result.yes).toBe("bigint");
+        expect(typeof proposal.result.no).toBe("bigint");
+        expect(typeof proposal.result.abstain).toBe("bigint");
+        // token
+        expect(typeof proposal.token.name).toBe("string");
+        expect(typeof proposal.token.symbol).toBe("string");
+        expect(typeof proposal.token.decimals).toBe("number");
+        expect(typeof proposal.token.address).toBe("string");
+        expect(proposal.token.address).toMatch(/^0x[A-Fa-f0-9]{40}$/i);
+      }
+    });
+    it("Should get a list of proposals from a specific dao", async () => {
+      const context = new ContextPlugin(contextParams);
+      const client = new ClientErc20(context);
+      const limit = 5;
+      const address = "0x663ac3c648548eb8ccd292b41a8ff829631c846d";
+      const params: IProposalQueryParams = {
+        limit,
+        sortBy: ProposalSortBy.CREATED_AT,
+        direction: SortDirection.ASC,
+        daoAddressOrEns: address,
+      };
+      const proposals = await client.methods.getProposals(params);
+
+      expect(Array.isArray(proposals)).toBe(true);
+      expect(proposals.length > 0 && proposals.length <= limit).toBe(true);
+    });
+    it("Should get a list of proposals from a dao that has no proposals", async () => {
+      const context = new ContextPlugin(contextParams);
+      const client = new ClientErc20(context);
+      const limit = 5;
+      const address = "0x1234567890123456789012345678901234567890";
+      const params: IProposalQueryParams = {
+        limit,
+        sortBy: ProposalSortBy.CREATED_AT,
+        direction: SortDirection.ASC,
+        daoAddressOrEns: address,
+      };
+      const proposals = await client.methods.getProposals(params);
+
+      expect(Array.isArray(proposals)).toBe(true);
+      expect(proposals.length === 0).toBe(true);
+    });
+    it("Should get a list of proposals from an invalid address", async () => {
+      const context = new ContextPlugin(contextParams);
+      const client = new ClientErc20(context);
+      const limit = 5;
+      const address = "0xn0tv4l1d";
+      const params: IProposalQueryParams = {
+        limit,
+        sortBy: ProposalSortBy.CREATED_AT,
+        direction: SortDirection.ASC,
+        daoAddressOrEns: address,
+      };
+      await expect(() => client.methods.getProposals(params)).rejects.toThrow(
```

Comment:
Ignorant question, is this extra rejects needed?

```suggestion
      await expect(() => client.methods.getProposals(params)).toThrow(
```

Reply:
Yes, if it is a promise, you need the `rejects`.